app/src/main/java/org/mozilla/tv/firefox/IntentValidator.kt

```diff
@@ -57,10 +60,18 @@ object IntentValidator {
 
         when (intent.action) {
             Intent.ACTION_MAIN -> {
-                val dialParams = intent.extras?.getString(DIAL_PARAMS_KEY) ?: return null
-                if (dialParams.isNotEmpty()) {
+                val extras = intent.extras ?: return null
+
+                if (extras.keySet().contains(DIAL_PARAMS_KEY)) {
```

Comment:
nit: `extras.containsKey` is more efficient. However, since we sometimes need to fetch the parameter anyway, it's slightly more efficient and has less duplication to just fetch it and check for null. However, it's less explicit. I prefer to fetch directly but I could go either way.